pkg/apiserver/apis/v1alpha3/ipblockusage_types.go

```diff
@@ -0,0 +1,68 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	GroupName    = "nsx.vmware.com"
+	Version      = "v1alpha3"
+	ResourceName = "ipblockusages"
+)
+
+const (
+	External Visibility = "EXTERNAL"
+	Project  Visibility = "PROJECT"
+	Private  Visibility = "PRIVATE"
+)
+
+type Visibility string
+
+type IPAddress struct {
+	Address string `json:"address,omitempty" protobuf:"bytes,1,opt,name=address"`
+	Subnet  string `json:"subnet,omitempty" protobuf:"bytes,2,opt,name=subnet"`
+}
+
+type AllocatedByVPC struct {
+	Count       int         `json:"count,omitempty" protobuf:"bytes,1,opt,name=count"`
+	IPAddresses []IPAddress `json:"ipAddresses,omitempty" protobuf:"bytes,2,opt,name=ipAddresses"`
+	CIDR        string      `json:"cidr,omitempty" protobuf:"bytes,3,opt,name=cidr"`
+	Visibility  Visibility  `json:"visibility,omitempty" protobuf:"bytes,4,opt,name=visibility"`
+}
+
+// IPBlocks contains the used IPAddresses and CIDR.
+type IPBlocks struct {
+	Path      string `json:"path,omitempty" protobuf:"bytes,1,rep,name=path"`
```

Comment:
Question is the IPBlock name unique? Finally, we may expose IPBlock as a K8s resource from CCI apiserver, and so using the K8s IPBlock resource name makes more sense.

Reply:
I do not mean your patch, but finally we need to expose IPBlocks in CCI apiserver too, so Tenant Admin can for example view External IPBlocks shared to the Project, and create Project IPBlocks.
In such cases, do all IPBlocks inc. shared External IPBlocks, Project IPBlocks, and Private IPBlocks have a unique name in NSX API inside a Project? What you would use for IPBlock resource names in CCI apiserver? We should use that name here, not NSX path, which is useless for CCI/Supervisor users.